File: 2522-partition-string-into-substrings-with-values-at-most-k/2522-partition-string-into-substrings-with-values-at-most-k.py
'''
Iterate through each digit in s. Add it as one's place to cur number.

Compare new cur with k, if larger then increment ans and reset cur to be digit

Return ans
'''

class Solution:
    def minimumPartition(self, s: str, k: int) -> int:
        cur,ans = 0, 1
        for d in s:
            if int(d) > k: return -1
            cur = 10*cur+int(d)
            if cur > k:
                ans+=1
                cur = int(d)
        return ans

File: 2522-partition-string-into-substrings-with-values-at-most-k/2522-partition-string-into-substrings-with-values-at-most-k.java
class Solution {
    public int minimumPartition(String s, int k) {
        int ans = 1;
        long cur = 0;
        for (int i = 0; i < s.length(); i++) {
            int digit = s.charAt(i) - '0';
            long tmp = cur * 10 + (long) digit;
            if (tmp <= (long) k) {
                cur = tmp;
            } else {
                if (digit > k) {
                    return -1;
                }
                ans++;
                cur = (long) digit;
            }
        }
        return ans;

    }
}